src/modules/cpu/kernel/transpose_generic.hpp

```diff
@@ -0,0 +1,482 @@
+{
+    Rpp32u alignedRows = (height / 4) * 4;
+    Rpp32u alignedCols = (width / 8) * 8;
+    Rpp32u vectorIncrement = 8;
+    Rpp32u dstRowVectorStride = vectorIncrement * dstRowStride;
+
+    int i = 0;
+    for(; i < alignedRows; i += 4)
+    {
+        Rpp32s k = (i / 4);
+        Rpp32f *srcPtrRow[4], *dstPtrRow[8];
+
+        for(int j = 0; j < 4; j++)
+            srcPtrRow[j] = srcPtrTemp + (i + j) * srcRowStride;
+        for(int j = 0; j < 8; j++)
+            dstPtrRow[j] = dstPtrTemp + j * dstRowStride + k * 4;
+
+        Rpp32u vectorLoopCount = 0;
+        for(; vectorLoopCount < alignedCols; vectorLoopCount += vectorIncrement)
+        {
+            __m256 pSrc[4];
+            pSrc[0] = _mm256_loadu_ps(srcPtrRow[0]);
+            pSrc[1] = _mm256_loadu_ps(srcPtrRow[1]);
+            pSrc[2] = _mm256_loadu_ps(srcPtrRow[2]);
+            pSrc[3] = _mm256_loadu_ps(srcPtrRow[3]);
+
+            __m128 pDst[8];
+            rpp_transpose4x8_avx(pSrc, pDst);
+            _mm_storeu_ps(dstPtrRow[0], pDst[0]);
+            _mm_storeu_ps(dstPtrRow[1], pDst[1]);
+            _mm_storeu_ps(dstPtrRow[2], pDst[2]);
+            _mm_storeu_ps(dstPtrRow[3], pDst[3]);
+            _mm_storeu_ps(dstPtrRow[4], pDst[4]);
+            _mm_storeu_ps(dstPtrRow[5], pDst[5]);
+            _mm_storeu_ps(dstPtrRow[6], pDst[6]);
+            _mm_storeu_ps(dstPtrRow[7], pDst[7]);
+
+            srcPtrRow[0] += vectorIncrement;
+            srcPtrRow[1] += vectorIncrement;
+            srcPtrRow[2] += vectorIncrement;
+            srcPtrRow[3] += vectorIncrement;
+            dstPtrRow[0] += dstRowVectorStride;
+            dstPtrRow[1] += dstRowVectorStride;
+            dstPtrRow[2] += dstRowVectorStride;
+            dstPtrRow[3] += dstRowVectorStride;
+            dstPtrRow[4] += dstRowVectorStride;
+            dstPtrRow[5] += dstRowVectorStride;
+            dstPtrRow[6] += dstRowVectorStride;
+            dstPtrRow[7] += dstRowVectorStride;
+        }
+    }
+
+    // Handle remaining cols
+    for(int k = 0; k < alignedRows; k++)
+    {
+        Rpp32f *srcPtrRowTemp = srcPtrTemp + k * srcRowStride + alignedCols;
+        Rpp32f *dstPtrRowTemp = dstPtrTemp + alignedCols * dstRowStride + k;
+        for(int j = alignedCols; j < width; j++)
+        {
+            *dstPtrRowTemp = *srcPtrRowTemp;
+            srcPtrRowTemp++;
+            dstPtrRowTemp += dstRowStride;
+        }
+    }
+
+    // Handle remaining rows
+    for( ; i < height; i++)
+    {
+        Rpp32f *srcPtrRowTemp = srcPtrTemp + i * srcRowStride;
+        Rpp32f *dstPtrRowTemp = dstPtrTemp + i;
+        for(int j = 0; j < width; j++)
+        {
+            *dstPtrRowTemp = *srcPtrRowTemp;
+            srcPtrRowTemp++;
+            dstPtrRowTemp += dstRowStride;
+        }
+    }
+}
+
+template<typename T>
+void transpose(T *dst, Rpp32u *dstStrides, T *src, Rpp32u *srcStrides, Rpp32u *dstShape, Rpp32u nDim)
```

Comment:
Probably rename to something like nd_transpose_generic_recursive()?

Reply:
Renamed the function to transpose_generic_nd_recursive()